src/adagrams.js

```diff
@@ -1,15 +1,133 @@
 export const drawLetters = () => {
-  // Implement this method for wave 1
+  const numTiles = 10;
+  const lettersInHand = [];
+  const currentLetterPool = [...listOfLetters]
+  for (let i = 0; i < numTiles; i++) {
+    // 98 letters, i want index from 0 to 97 inclusive
+    let randIndex = Math.floor(Math.random() * currentLetterPool.length);
+    let randLetter = currentLetterPool[randIndex];
+    lettersInHand.push(randLetter);
+    currentLetterPool.splice(randIndex, 1)
+  }
+  return lettersInHand
 };
 
 export const usesAvailableLetters = (input, lettersInHand) => {
-  // Implement this method for wave 2
+  const letterMap = lettersInHandToDict(lettersInHand);
+  input = input.toUpperCase();
+  for (const char of input) {
+    // doesn't have char in map -> false
+    // has char in map
+      // if value > 0 -> -1
+      // if value = 0 -> false
+    if (!letterMap.has(char)) {
+      return false;
+    } else if (letterMap.has(char)) {
+      if (letterMap.get(char) > 0) {
+        letterMap.set(char, letterMap.get(char) - 1);
+      } else {
+        return false;
+      }
+    }
+  }
+  // if it hasn't return false at this point, return true
+  return true;
```

Comment:
I couldn't quite figure out why the `else` condition on lines 28-29 were needed. I had to debug a test to see that it handles a scenario where a word repeats a letter but the `letterMap` doesn't have that many occurrences of the letter (ex: word is "good" but hand only has one of each g, o, d.

We can simplify the logic a little:

```suggestion
    if (letterMap.has(char) && letterMap.get(char) > 0) {
        letterMap.set(char, letterMap.get(char) - 1);
    } else {
      return false;
    }
    
  return true;
```